File: julia/secret-handshake/secret-handshake.jl
function secret_handshake(code)
    code_binary = vcat(digits(code, base=2), [0, 0, 0, 0, 0])
    responses = ["wink", "double blink", "close your eyes", "jump"]
    response = []
    for i in 1:4
        if code_binary[i] == 1
            push!(response, responses[i])
        end
    end
    code_binary[5] == 1 ? reverse(response) : response
end

File: r/secret-handshake/secret-handshake.R
handshake <- function(n) {
  if (n %% 16 == 0) return(c())
  
  responses  <- c('wink', 'double blink', 'close your eyes', 'jump')
  actions <- responses[bitwAnd(2^(0:3), n) != 0]
  if (bitwAnd(2^4, n) == 0) actions
  else rev(actions)
}
